virtualizarr/writers/icechunk.py

```diff
@@ -323,17 +325,22 @@ def write_manifest_virtual_refs(
         op_flags=[["readonly"]] * 3,  # type: ignore
     )
 
+    virtual_chunk_spec_list = []
```

Comment:
This builds up a list of `VirtualChunkSpec` objects then does a single `set_virtual_refs`. Should this be batched or handled differently?

Reply:
I think this is what we want to do in this PR at least. We will gain the benefit of not going back and forth between python and rust on every chunk, only on each array.

Also you could use a list comprehension instead of creating an empty list and then appending to it

Reply:
Good call!